### PR TITLE
feat: thanos support

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

- Add Thanos resource (`aiven_thanos`) to allow for the creation and management of Thanos services

## Why this way

🚀

### CODE REVIEW

**Review by byashimov — CHANGES_REQUESTED** [review]

File: internal/sdkprovider/service/thanos/thanos.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+)
+
+func thanosSchema() map[string]*schema.Schema {
+	return schemautil.ServiceCommonSchemaWithUserConfig(schemautil.ServiceTypeThanos)
+}
+
+func ResourceThanos() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Creates and manages an [Aiven for Metrics®](https://aiven.io/docs/products/metrics/concepts/metrics-overview) service.",
```
[the remark this answers is not in the document]
Comment (Serpentiel) [reply internal/sdkprovider/service/thanos/thanos.go]: I remember doing that. No idea why it's not there. I'll update the PR.

**Comment by byashimov** [comment]
I have created Thanos in the Console, it has connection info fields:
image

I believe we should proxy them to the resource.

**Comment by Serpentiel** [comment]
Temporarily marked as draft. Some things should be clarified.

File: internal/sdkprovider/service/thanos/thanos_test.go [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    value = "val"
+  }
+
+  tech_emails {
+    email = "techsupport@company.com"
+  }
+
+  thanos_user_config {
+    query_frontend {
+      query_range_align_range_with_step = true
```
Comment (byashimov) [inline internal/sdkprovider/service/thanos/thanos_test.go:95]: true is default value, might give a false positive

**Review by byashimov — APPROVED** [review]
lgtm

**Pull request merged**